OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/AmenityMenuBuilder.java

```diff
@@ -877,12 +896,53 @@ public void onClick(View view) {
 		return new CollapsableView(view, this, collapsed);
 	}
 
+	private String getSocialMediaUrl(String key, String value) {
+		// Regex for url without protocol, with at least 2 valid domains, slash and at least one
+		// char after it (e.g. a-c.com/a)
+		String urlMask = "[a-zA-Z0-9]+[-]*[a-zA-Z0-9]+(\\.[a-zA-Z0-9]+[-]*[a-zA-Z0-9]+)+\\/.+";
+
+		if (Pattern.matches(urlMask, value)) {
```

Comment:
is it needed at all? Are there real examples of such broken data?
Looks like unnecessary complication - btw there are ready to use URL parser new URL()

Reply:
I think this check is reasonable. From my POV this is the only way to distinguish username (and what looks like it) and url without protocol.

Reply:
user name can't contain '/' as I mentioned previously